File: Homework_2/app/src/main/java/com/example/homework_2/models/U1.java
package com.example.homework_2.models;

public class U1 extends Rocket{

    public U1 (){
        setCost(100);
        setWeight(10000);
        setMaxWeight(18000);
        setCurrentWeight(getWeight());
    }

    @Override
    public boolean land(){
        int random = (int)(Math.random() * 100 + 1);
        return 1.0 * (getCurrentWeight() - getWeight()) / (getMaxWeight() - getWeight()) <= random;
    }

    @Override
    public boolean launch(){
        int random = (int)(Math.random() * 100 + 1);
        return 5.0 * (getCurrentWeight() - getWeight()) / (getMaxWeight() - getWeight()) <= random;
    }
}

File: SpaceShip_Kotlin/app/src/main/java/com/example/spaceship_kotlin/moduls/U1.kt
package com.example.spaceship_kotlin.moduls

class U1: Rocket() {
    init {
        cost = 100
        weight = 10000
        maxWeight =18000
        currentWeight = weight
    }

    override fun land(): Boolean{
        val random = (Math.random() * 100 + 1).toInt()
        return 1.0 * (currentWeight - weight) / (maxWeight - weight) <= random
    }

    override fun launch(): Boolean {
        val random = (Math.random() * 100 + 1).toInt()
        return 5.0 * (currentWeight - weight) / (maxWeight - weight) <= random
    }


}
